### ISSUE TITLE
Bad Main Class

### ISSUE THREAD

**Issue by fante76** [vaadin/testbench#960]
In file vaadin-testbench-standalone-5.0.0.jar the MANIFEST references a not existing class org.openqa.grid.selenium.GridLauncher instead the correct class org.openqa.grid.selenium.GridLauncherV3.
Replacing, make it run correctly.

**Comment by johannesh2** [comment]
Thank you. This was fixed in 5.1. Maybe it would've been worth to backport to 5.0 as it's such an easy change. 5.1 is pretty close to final. Have you had time to test 5.1.0.beta1?

**Comment by fante76** [comment]
Hi
We just started with the test project. I think it would be a good idea to
test beta version. I would scheduled it for monday. Thank you

Il 10 nov 2017 15:20, "Johannes Häyry" <notifications@github.com> ha
scritto:

Thank you. This was fixed in 5.1. Maybe it would've been worth to backport
to 5.0 as it's such an easy change. 5.1 is pretty close to final. Have you
had time to test 5.1.0.beta1?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/vaadin/testbench/issues/960#issuecomment-343485248>,
or mute
the thread

**Comment by fante76** [comment]
I've tried 5.1.0.beta1 but I have runtime error in this code:
`NotificationElement notification = $(NotificationElement.class).first();
notification.close();`
It does not find method `org.openqa.selenium.support.ui.WebDriverWait.until`
The code compile correctly. By now I need to switch back to 5.0.0
Sorry.

**Comment by alvarezguille** [comment]
Hi @fante76, could you check in your project what version is resolved for 
```
<groupId>com.google.guava</groupId>
<artifactId>guava</artifactId>
```
I have a suspicion that it'll be lower than 21.

One of the changes in Testbench 5.1 is update to Selenium 3.4.0 which according to [their changelog](https://raw.githubusercontent.com/SeleniumHQ/selenium/master/java/CHANGELOG) updated their `guava` dependency in 3.1.0. Forcing the resolution of guava to version 21.0 or higher should fix that issue.

**Comment by fante76** [comment]
I've tried again. Guava is wright to version 21. Running a test give me this error trying to close a notification.

`
java.lang.NoSuchMethodError: org.openqa.selenium.support.ui.WebDriverWait.until(Lcom/google/common/base/Function;)Ljava/lang/Object;
at com.vaadin.testbench.elements.NotificationElement.close(NotificationElement.java:41)
`

**Comment by alvarezguille** [comment]
The Bad Main Class error was fixed in #897 and released in [5.1.0](https://github.com/vaadin/testbench/releases/tag/5.1.0)

The `java.lang.NoSuchMethodError` looks like a version conflict. Feel free to create new issues if found with 5.1.0.
